### PR TITLE
Refactored code structure and implemented tests for all relevant graphql endpoints

### PR DESCRIPTION
**Pull request by LinusLjung**
Tests can be run with the `npm test` command

### CODE REVIEW

File: server/database.js [hunk 1]
```diff
@@ -26,12 +20,7 @@ class Database {
 
 	handleError = console.error.bind(console, 'connection error:'); // eslint-disable-line no-console
 
-	handleOpen = () => {
-		this.CitationModel = citationModel();
-		this.UserModel = userModel();
-		this.Reaction = reactionModel();
-		this.ReactionType = reactionTypeModel();
-	};
+	handleOpen = () => {};
```
Comment (sarjir) [inline server/database.js]: What does this function do? 😇

File: server/graphql/root-query.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+		console.error(e);
+		return false;
+	}
+}
+
+export default new GraphQLObjectType({
+	name: 'RootQueries',
+	fields: () => ({
+		echo: {
+			type: GraphQLString,
```
Comment (sarjir) [inline server/graphql/root-query.js:33]: Such large indententions, much why??

**Review by sarjir — COMMENTED** [review]
Eeek, so exited!! I don't understand much but I approve. I APPROOOOVE!! Can you please explain more the next time we meet?? 😁

**Review by sarjir — APPROVED** [review]

File: server/database.js (hunk 1 above)
Comment (LinusLjung) [reply server/database.js]: Not much ;)

Just a no-op, but I might as well remove it :)

File: server/graphql/root-query.js (hunk 2 above)
Comment (LinusLjung) [reply server/graphql/root-query.js:33]: Every indentation is only 1 tab character, but github interprets them as 8 spaces

File: package.json [hunk 3]
```diff
@@ -23,18 +23,24 @@
...
     "eslint-plugin-react": "^6.2.0",
     "less": "^2.7.1",
     "less-loader": "^2.2.3",
+    "mocha": "^3.0.2",
     "react": "^15.2.1",
     "react-dom": "^15.2.1",
     "react-redux": "^4.4.5",
     "redux": "^3.5.2",
+    "sinon": "^1.17.5",
+    "sinon-as-promised": "file:///Users/linus/projects/sinon-as-promised",
```
Comment (LinusLjung) [inline package.json]: Incorrect dependency link

File: server/graphql/root-query.js (hunk 2 above)
Comment (sarjir) [reply server/graphql/root-query.js:33]: Oh! I see! Did we decide on using tabs?

Comment (LinusLjung) [reply server/graphql/root-query.js:33]: I'm pretty sure we did? Have you been using spaces? :)
I don't mind either way

**Pull request merged**